src/App.tsx

```diff
@@ -1,39 +1,56 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
-
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+export class App extends React.Component {
+  state = {
+    clockName: 'Clock-0',
+    hasClock: true,
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    document.addEventListener('contextmenu', this.handleClockDissappear);
+    document.addEventListener('click', this.handleClockAppear);
+  }
+
+  handleClockDissappear = (event: MouseEvent) => {
+    event.preventDefault();
+    this.setState({ hasClock: false });
+    document.removeEventListener('contextmenu', this.handleClockDissappear);
```

Comment:
It is not necessary to remove events here
It would be enough to just add on mount and remove on unmount

Reply:
agree with [comment](https://github.com/mate-academy/react_clock/pull/1658/files#r1267813529)